tools/latency_logging/print_timestamps.py

```diff
@@ -0,0 +1,166 @@
+from tools.lib.route import Route
+from tools.lib.logreader import LogReader
+from tabulate import tabulate
+import json
+from collections import defaultdict
+
+r = Route("9f583b1d93915c31|2022-03-28--17-25-12") 
+lr = LogReader(r.log_paths()[0])
+
+services = ['camerad', 'modeld', 'plannerd', 'controlsd', 'boardd']
+
+msgq_to_service = {}
+msgq_to_service['roadCameraState'] = "camerad"
+msgq_to_service['wideRoadCameraState'] = "camerad"
+msgq_to_service['modelV2'] = "modeld"
+msgq_to_service['lateralPlan'] = "plannerd"
+msgq_to_service['longitudinalPlan'] = "plannerd"
+msgq_to_service['sendcan'] = "controlsd"
+
+service_to_durations = defaultdict(list)
+service_to_durations['camerad'] = ['timestampSof', 'processingTime']
+service_to_durations['modeld'] = ['modelExecutionTime', 'gpuExecutionTime']
+service_to_durations['plannerd'] = ["solverExecutionTime"]
+
+# Build translation dict
+translationdict = {}
+for msg in lr:
+  if msg.which() == "logMessage" and "translation" in msg.logMessage:
+    msg = msg.logMessage.replace("'", '"').replace('"{', "{").replace('}"', "}").replace("\\", '')
```

Comment:
looks like this is just a string-ified python dict instead of properly serialized json

Reply:
When printing msg:

( logMonoTime = 599335952933494,                                                                                            |i
  logMessage = "{\"created\": 1648513515.2507977, \"ctx\": {\"daemon\": \"camerad\", \"device\": \"tici\", \"dirty\": true, |o
ongle_id\": \"9f583b1d93915c31\", \"version\": \"0.8.14\"}, \"filename\": \"selfdrive/camerad/cameras/camera_qcom2.cc\", \"f|n
name\": \"process_road_camera\", \"levelnum\": 10, \"lineno\": 1035, \"msg\": \"{\\\"timestamp\\\": {\\\"event\\\": \\\"Road|s
era: Image set\\\", \\\"time\\\": \\\"599335951599136\\\"}}\"}",                                                            |
  valid = true )

could look into where the logs are set, but it is not only the logging which I have created

Reply:
The cpp json11 lib causes this. Confused why it does that, using it exactly as the docs/

\"{\\\"timestampExtra\\\"|        'longitudinalPlan':'plannerd',
: {\\\"event\\\": \\\"Pipeline end\\\", \\\"i                                     |        'sendcan':'controlsd'
nfo\\\": {\\\"logMonoTime\\\": \\\"599335735751377\\\"}, \\\"time\\\": \\\"5993357|        }
36414806\\\"}}\"}"